openpype/pipeline/publish/lib.py

```diff
@@ -623,12 +641,58 @@ def get_instance_staging_dir(instance):
     Returns:
         str: Path to staging dir of instance.
     """
+    staging_dir = instance.data.get('stagingDir')
+    if staging_dir:
+        return staging_dir
+
+    openpype_temp_dir = os.getenv("OPENPYPE_TEMP_DIR")
+    custom_temp_dir = None
+    if openpype_temp_dir:
+        if "{" in openpype_temp_dir:
+            custom_temp_dir = _formated_staging_dir(
+                instance, openpype_temp_dir
+            )
+        elif os.path.exists(openpype_temp_dir):
+            custom_temp_dir = openpype_temp_dir
 
-    staging_dir = instance.data.get("stagingDir")
-    if not staging_dir:
+
+    if custom_temp_dir:
+        staging_dir = os.path.normpath(
+            tempfile.mkdtemp(
+                prefix="pyblish_tmp_",
+                dir=custom_temp_dir
+            )
+        )
+    else:
         staging_dir = os.path.normpath(
             tempfile.mkdtemp(prefix="pyblish_tmp_")
         )
-        instance.data["stagingDir"] = staging_dir
+    instance.data['stagingDir'] = staging_dir
+
+    instance.context.data["cleanupFullPaths"].append(staging_dir)
 
     return staging_dir
+
+
+def _formated_staging_dir(instance, openpype_temp_dir):
```

Comment:
Currently has a typo in function name. `formated` -> `formatted`

I actually think however, we can also just refactor this to `_format_staging_dir`?